### DISCUSSION TITLE
Access model in custom page view

### DISCUSSION THREAD

**Discussion by kvostyc** [filamentphp/filament#6438] [Help] [answered]
Hey, i got my own action in resource and custom page. But i need to know how to access this model

In resource

``
actions([
   Tables\Actions\EditAction::make(),
                Tables\Actions\DeleteAction::make(),
                Action::make('Reply')
                    ->url(fn (OrderPending $record): string => url('admin/order-pendings/'. $record->id .'/reply'))->openUrlInNewTab()->color('success')->icon('heroicon-o-clipboard')
            ])``

`    public static function getPages(): array
    {
        return [
            'index' => Pages\ListOrderPendings::route('/'),
            'create' => Pages\CreateOrderPending::route('/create'),
            'edit' => Pages\EditOrderPending::route('/{record}/edit'),
            'reply' => Pages\ReplyPending::route('/{record}/reply')
        ];
    }`

**Comment by danharrin** [comment, marked as answer]
Firstly, generate URLs like this: `OrderPendingResource::getUrl('reply' => ['record' => $record])`

Inside the `ReplyPending` class, define a `$record` property. Also define a `mount()` method:

```php
public $record;

public function mount($record): void
{
    $this->record = OrderPending::find($record);
}
```

Then, you can just access the record anywhere in the view or class with `$this->record`
